commit 1ecd68dc1e0bc7216f507058f19738e3c109e594
Author: Pawel MOLL <pawel.moll@st.com>
Date:   Mon Jan 28 16:23:54 2008 +0000

    Fix for SMSC911x driver compilation with CONFIG_SMSC911x_DMA_2D (bug #3335).
    
    Signed-off-by: Pawel MOLL <pawel.moll@st.com>

diff --git a/drivers/net/smsc_911x/st40-shdma.c b/drivers/net/smsc_911x/st40-shdma.c
index 6626f96..3aec259 100644
--- a/drivers/net/smsc_911x/st40-shdma.c
+++ b/drivers/net/smsc_911x/st40-shdma.c
@@ -65,7 +65,6 @@ DWORD Platform_RequestDmaChannelSg(
 	dma_reqs[0] = dma_req_config(chan, dma_req_lines[0], &dma_req_configs[0]);
 	dma_reqs[1] = dma_req_config(chan, dma_req_lines[1], &dma_req_configs[1]);
 
-printk("%s: req %x and %x\n", __FUNCTION__, dma_reqs[0], dma_reqs[1]);
 	return chan;
 }
 
@@ -75,6 +74,9 @@ static void Platform_ReleaseDmaChannel_sg(void)
 
 	for(i=0;i<MAX_NODELIST_LEN;i++)
 		dma_params_free(&rx_transfer_paced[i]);
+
+	dma_req_free(dwDmaChannel, dma_reqs[0]);
+	dma_req_free(dwDmaChannel, dma_reqs[1]);
 }
 
 static void Platform_DmaInitialize_sg(void)
@@ -152,8 +154,6 @@ void Platform_ReleaseDmaChannel(
 	DWORD dwDmaChannel)
 {
 	free_dma(dwDmaChannel);
-	dma_req_free(dwDmaChannel, dma_reqs[0]);
-	dma_req_free(dwDmaChannel, dma_reqs[1]);
 	dma_params_free(&tx_transfer);
 	Platform_ReleaseDmaChannel_sg();
 }
